modules/imap/functions.php

```diff
@@ -1728,3 +1732,39 @@ function sortCombinedMessages($list, $sort) {
 
     return $list;
 }
+
+if (!hm_exists('save_sent_msg')) {
+function save_sent_msg($handler, $imap_id, $mailbox, $imap_details, $msg, $msg_id, $show_errors = true) {
+    $specials = get_special_folders($handler, $imap_id);
+    if (array_key_exists('sent', $specials) && $specials['sent']) {
+        $sent_folder = $specials['sent'];
+    }
+
+    if (!$sent_folder) {
+        $auto_sent = $mailbox->get_special_use_mailboxes('sent');
+        if (!array_key_exists('sent', $auto_sent)) {
+            return;
+        }
+        $sent_folder = $auto_sent['sent'];
+    }
+    if (!$sent_folder) {
+        Hm_Debug::add(sprintf("Unable to save sent message, no sent folder for server %s %s", $mailbox->server_type(), $imap_details['server']));
+    }
+    $uid = null;
+    if ($sent_folder) {
+        Hm_Debug::add(sprintf("Attempting to save sent message for server %s in folder %s", $mailbox->server_type(), $imap_details['server'], $sent_folder));
+        if (! $mailbox->store_message($sent_folder, $msg)) {
```

Comment:
store message returns the UID, no need to get messages sorting by arrival and get it by the ID just to return the UID.

Reply:
This was already solved. I think a wrong rebase deleted it.